Q-8.
The anxiety levels of 10 participants were measured before and after a new therapy. The scores are not normally distributed. Use the Wilcoxon signed-rank test to test whether the therapy had a significant effect on anxiety levels.

In [1]:
import pandas as pd
import scipy.stats as stats

In [2]:
#Reading the data
data={'Participant':[1,2,3,4,5,6,7,8,9,10],
                  'Before_therapy':[10,8,12,15,6,9,11,7,14,10],
                  'After_therapy':[7,6,10,12,5,8,9,6,12,8]}
df = pd.DataFrame(data)
df['Difference'] = df.Before_therapy - df.After_therapy

In [3]:
df.head()

Participant  Before_therapy  After_therapy  Difference
0            1              10              7           3
1            2               8              6           2
2            3              12             10           2
3            4              15             12           3
4            5               6              5           1

In [4]:
statistic, p_value = stats.wilcoxon(df.Difference)

print("Wilcoxon signed-rank test:")
print("Statistic:", statistic)
print("p-value:", p_value)

Wilcoxon signed-rank test:
Statistic: 0.0
p-value: 0.001953125


Consdering significance level of 0.05, since we got the p-value (0.001953125) which is less than 0.05, we can conclude that there is strong evidence to reject the null hypothesis. This means that the therapy had a significant effect on anxiety levels.



In other words, the results suggest that there is a significant difference between the anxiety levels before and after the therapy.